src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -208,6 +211,45 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         });
     }
 
+    // Function Description:
+    // - Static hepler for building a new TermControl from an already existing
+    //   content. We'll attach the existing swapchain to this new control's
+    //   SwapChainPanel. The IKeyBindings might belong to a non-agile object on
+    //   a new thread, so we'll hook up the core to these new bindings.
+    // Arguments:
+    // - content: The pre-existing ControlInteractivity to connect to.
+    // - keybindings: The new IKeyBindings instance to use for this control.
+    // Return Value:
+    // - The newly constructed TermControl.
+    Control::TermControl TermControl::AttachContent(Control::ControlInteractivity content,
+                                                    const Microsoft::Terminal::Control::IKeyBindings& keyBindings)
+    {
+        const auto term{ winrt::make_self<TermControl>(content) };
+        term->_AttachDxgiSwapChainToXaml(reinterpret_cast<HANDLE>(term->_core.SwapChainHandle()));

```

Comment:
A swap chain doesn't necessarily exist in the renderers when you call `SwapChainHandle()` and you'll have to work around this by making sure to hook up the `SwapChainChanged` handlers before running this code. In other words, we need events for swap chain changes before getting the current swap chain, because we might otherwise miss a swap chain change.

Reply:
(I moved where we set up the callback into the ctor)